ims/einstein/operations.py

```diff
@@ -635,3 +639,31 @@ def remake_mappings(self):
             logger.exception('')
         except NotImplementedError:
             pass
+
+    @log
+    def run_script(self, img, script):
+        try:
+            if not self.is_admin:
+                raise AuthorizationFailedException()
+            directory = "/tmp/script/"
+            os.makedirs(directory)
+            script_path = directory + "/test_script.sh"
+            ret = self.fs.map(img)
+            decode_script = base64.b64decode(script)
+            with open(script_path, "w") as write_script:
+                write_script.write(decode_script)
+            os.chmod(script_path, S_IRWXU)
+            p = subprocess.Popen(script_path + " " + ret, shell=True,
```

Comment:
This shell=True is dangerous. Also we are just mapping the image, but not mounting it on the VM due to which the script will not be able to access the contents of the image.

The right way to do it would be to mount the image using the "mount" command in some location in /tmp, chroot to that location and then run the script.

Right now if the user is admin he can technically run any script which can be "rm -rf /" which is dangerous. The chrooting will help with this, but might delete the entire image. So it would be better to clone the image before mapping and deleting it at the end.